tests/lib/fakestore/store/store.go

```diff
@@ -257,6 +259,63 @@ func (s *Store) searchEndpoint(w http.ResponseWriter, req *http.Request) {
 	fmt.Fprintf(w, "search not implemented")
 }
 
+func (s *Store) repairsEndpoint(w http.ResponseWriter, req *http.Request) {
+	brandAndRepairID := strings.Split(strings.TrimPrefix(req.URL.Path, "/v2/repairs/"), "/")
+	if len(brandAndRepairID) != 2 {
+		http.Error(w, "missing brand and repair ID", 400)
+		return
+	}
+
+	bs, err := s.collectAssertions()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("internal error collecting assertions: %v", err), 500)
+		return
+	}
+
+	a, err := s.retrieveAssertion(bs, asserts.RepairType, brandAndRepairID)
+	if asserts.IsNotFound(err) {
+		w.Header().Set("Content-Type", "application/problem+json")
+		w.WriteHeader(404)
+		w.Write([]byte(`{"status": 404}`))
+		return
+	}
+	if err != nil {
+		http.Error(w, fmt.Sprintf("cannot retrieve repair assertion %v: %v", brandAndRepairID, err), 400)
+		return
+	}
+
+	// there are two cases, one where we are asked for the full assertion, and
+	// one where we are asked for JSON headers of the assertion, so check which
+	// one we were asked for by inspecting the Accept header
+
+	// TODO: what about the If-None-Match header?
```

Comment:
it will contain the revision in quotes,  if the revision matches the revision of  a, then this can return a 304

Reply:
ok, I added support for it, I'm not really sure how to test the code in fakestore for that in the spread tests... we also have been fairly bad about writing unit tests for the fakestore proper :-(